### DISCUSSION TITLE
Add support for ownCloud News / Tiny RSS / Newsblur

### DISCUSSION THREAD

**Discussion by josh64x2** [ViennaRSS/vienna-rss#345] [Ideas]
Originally requested [by Northdude from the forums](http://www.cocoaforge.com/viewtopic.php?f=18&t=26635&sid=094ba2f2875a499d915c42a6dd454e0b).

Many people are already using ownCloud, but there are just mobile applications available to sync with the integrated News App. Some of this mobile applications are Open Source, maybe it would not be that hard to modify the existing code so that Vienna could support it. [They are listed here, subitem news](https://github.com/owncloud/core/wiki/Apps).

[ownCloud News API](https://github.com/owncloud/news/wiki/API-1.2)

**Comment by barijaona** [comment]
We should also consider [Tiny Tiny RSS](http://tt-rss.org/redmine/projects/tt-rss/wiki) (aka TT-RSS), which is another lightweight cloud-based news aggregator  : in my remembrance, it was easier to install and to configure than ownCloud.

Unfortunately, most TT-RSS clients are developed under a GNU license.
But [newsbeuter](https://github.com/akrennmair/newsbeuter/) is governed by an MIT license, and supports Tiny Tiny RSS as well as some Open Reader servers like [The Old Reader and FeedHQ, plus Newsblur](http://www.newsbeuter.org/doc/newsbeuter.html#_the_old_reader_support). So examining its code might be interesting.

[Tiny Tiny RSS API](http://tt-rss.org/redmine/projects/tt-rss/wiki/JsonApiReference).

**Comment by lenucksi** [comment]
:+1: I would also appreciate Newsblur support very much.

**Comment by barijaona** [comment]
An interesting alternative for the apps listed here is [FreshRSS](
https://github.com/FreshRSS/FreshRS)S, which is open source, lightweight, uses the OpenReader consensus and [is being fixed to be fully compatible with Vienna](https://github.com/FreshRSS/FreshRSS/issues/2091).
.

**Comment by rodneyrod** [comment]
It's been brought up that the OpenReader API is incompatible with how Nextcloud authenticates, so unless changes are made to Vienna to support how Nextcloud does their login, or if that was worked into the OpenReader spec, this doesn't look like it will be supported.

Link to discussion:
https://github.com/nextcloud/news/issues/618#issuecomment-574452287

**Comment by yougotwill** [comment]
Can anyone help me with this. Been using Vienna for a few years but just setup a tiny-rss instance on a raspberry pi and would love to integrate them? 🙏

**Reply by barijaona** [reply]
I have looked into tt-rss at a time, but I found it a bit challenging to integrate an additional API into Vienna.
Have you looked at installing FreshRSS on your Raspberry Pi ? My personal workflow integrates Vienna and a free public FreshRSS instance (with other feeds that I fetch directly from Vienna only), and it works well.

**Reply by yougotwill** [reply]
Hi, thanks for replying. I haven't tried FreshRSS. I'll look into it today. Thanks for the recommendation 🙏

**Comment by barijaona** [comment]
My recommendation for people liking Vienna and interested in self hosting is [FreshRSS](https://freshrss.org).

Connecting Vienna (or another compatible application) to your FreshRSS instance is easy, once you understand the following precautions:
- you have to specify a specific password (different from the one you use to access the web interface), through a web page: Settings -> Profile -> API Management;  
If you use a password management tool, be careful not to mix up the two passwords (Web interface password and API password).
- you have to provide to the client application (for instance Vienna) : 
    - an URL pointing to your server: most of the time it is something similar to _http(s)://example.org/**api/greader.php**_ , but a few applications (especially ReadKit) require the base URL _http(s)://example.org/_
    - your username
    - the API password mentioned above.

Besides Vienna, I have successfully tested FreshRSS with the following iOS clients (listed without any specific order):
- [Reeder Classic](https://www.reederapp.com/classic/) (paid)
- [Fluent Reader Lite](https://hyliu.me/fluent-reader-lite/) (free)
- [Fiery Feeds](https://voidstern.net/fiery-feeds) (free, premium subscription is not necessary)
- [NetNewsWire](https://netnewswire.com) (free, open source)
- [ReadKit](https://readkit.app) (free, premium subscription is not necessary)

A list of other possible clients (some for Android, Windows, Linux, FreeBSD) is available here : https://github.com/FreshRSS/FreshRSS/blob/edge/README.md#apis--native-apps
